tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/RepositoryScriptService.cs

```diff
@@ -0,0 +1,164 @@
+using System.Collections.Specialized;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Sdk.Tools.Cli.Helpers;
+
+namespace Azure.Sdk.Tools.Cli.Services;
+
+public interface IRepositoryScriptService
+{
+    Task<string> GetCommand(string commandName, string packagePath, CancellationToken ct);
+    Task<bool> HasImplementation(string commandName, string packagePath, CancellationToken ct);
+    Task<(bool invoked, ProcessResult result)> TryInvoke(
+        string commandName,
+        string packagePath,
+        OrderedDictionary args,
+        bool invokeFromRepoRoot = true,
+        CancellationToken ct = default
+    );
+    Task<(bool invoked, ProcessResult result)> TryInvoke(string commandName, string packagePath, OrderedDictionary args, CancellationToken ct = default);
```

Comment:
What is the 'invoked' value for here? Wouldn't ProcessResult have enough information to know? 

(and if it doesn't, what were you thinking you'd do if it's false/true?)

Reply:
The value represents whether we actually found a script override. I haven't been able to think of a better name but agree `invoked` is not clear. Any ideas?

We could also return a null process result instead to indicate that we never tried to run anything and therefore have no process result? But I like the idea of having separate named returns vs. just having to know that we intend a null response to mean "we didn't find an override". So maybe `Task<(bool noOverrideFound, ProcessResult result)>`?

If it's true, then the caller is done and can return the process result. If it's false, then the caller can either execute the c#-based fallback implementation, throw a not implemented error, or return a no-op.